vpr/src/route/rr_graph_uxsdcxx_interface_impl.h

```diff
@@ -0,0 +1,1919 @@
+     *   <xs:sequence>
+     *     <xs:element maxOccurs="unbounded" name="meta" type="meta" />
+     *   </xs:sequence>
+     * </xs:complexType>
+     */
+    inline void* add_metadata_meta(void* data) override {
+        if (data != nullptr) {
+            auto* bind = static_cast<MetadataBind*>(data);
+            bind->assert_clear();
+            return data;
+        } else {
+            return nullptr;
+        }
+    }
+    inline void finish_metadata_meta(void* data) override {
+        if (data != nullptr) {
+            auto* bind = static_cast<MetadataBind*>(data);
+            bind->bind();
+        }
+    }
+
+    /** Generated for complex type "node_loc":
+     * <xs:complexType name="node_loc">
+     *   <xs:attribute name="xlow" type="xs:int" use="required" />
+     *   <xs:attribute name="ylow" type="xs:int" use="required" />
+     *   <xs:attribute name="xhigh" type="xs:int" use="required" />
+     *   <xs:attribute name="yhigh" type="xs:int" use="required" />
+     *   <xs:attribute name="side" type="loc_side" />
+     *   <xs:attribute name="ptc" type="xs:int" use="required" />
+     * </xs:complexType>
+     */
+    inline void set_node_loc_side(uxsd::enum_loc_side side, void* data) override {
+        int* inode = static_cast<int*>(data);
+        auto& node = (*rr_nodes_)[*inode];
+
+        switch (side) {
+            case uxsd::enum_loc_side::UXSD_INVALID:
+                // node_loc.side is only expected on IPIN/OPIN
+                VTR_ASSERT(!(node.type() == IPIN || node.type() == OPIN));
+                break;
+            case uxsd::enum_loc_side::LEFT:
+                node.set_side(LEFT);
+                break;
+            case uxsd::enum_loc_side::RIGHT:
+                node.set_side(RIGHT);
+                break;
+            case uxsd::enum_loc_side::TOP:
+                node.set_side(TOP);
+                break;
+            case uxsd::enum_loc_side::BOTTOM:
+                node.set_side(BOTTOM);
+                break;
+            default:
+                VPR_FATAL_ERROR(VPR_ERROR_OTHER,
+                                "Invalid side %d", side);
+        }
+    }
+
+    inline const char* get_meta_name(void* data) override {
+        auto* meta_value = static_cast<
+            std::pair<
+                std::string,
+                std::vector<t_metadata_value>>*>(data);
+        return meta_value->first.c_str();
+    }
+    inline const char* get_meta_value(void* data) override {
+        auto* meta_value = static_cast<
+            std::pair<
+                std::string,
+                std::vector<t_metadata_value>>*>(data);
+        VTR_ASSERT(meta_value->second.size() == 1);
+        return meta_value->second[0].as_string().c_str();
+    }
+    inline size_t num_metadata_meta(void* data) override {
+        auto* meta = static_cast<struct t_metadata_dict*>(data);
+        return meta->size();
+    }
+    inline void* get_metadata_meta(int n, void* data) override {
```

Comment:
Maybe it's better to generate a `begin/next/end` set of functions in uxsdcxx? We would need one `*data` for the iterator state and one `*data` for the parent state.